### PR TITLE
Add WebdriverIO test reports

### PR DESCRIPTION
**Pull request by Fatxx**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Enables generation of test reports for cucumber e2e tests, these reports are created as HTML pages under `wdio/reports`.

**TODO:** Where should we host these reports?

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses [#190](https://app.zenhub.com/workspaces/mobile-automation-board-633766327776c3e8970e0a94/issues/metamask/mobile-planning/190)

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by sethkfman — APPROVED** [review]
LGTM

File: app/components/Views/AccountBackupStep1/index.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import Engine from '../../../core/Engine';
 import { ONBOARDING_WIZARD } from '../../../constants/storage';
 import { CHOOSE_PASSWORD_STEPS } from '../../../constants/onboarding';
 import SkipAccountSecurityModal from '../../UI/SkipAccountSecurityModal';
-import SeedPhraseVideo from '../../UI/SeedPhraseVideo';
+// import SeedPhraseVideo from '../../UI/SeedPhraseVideo';
```
Comment (cortisiko) [inline app/components/Views/AccountBackupStep1/index.js:27]: Is this disabling for QA builds as well as our regular builds?

File: app/components/Views/AccountBackupStep1/index.js [hunk 2]
```diff
@@ -215,6 +215,7 @@ const AccountBackupStep1 = (props) => {
 
   const hideWhatIsSeedphrase = () => setWhatIsSeedphraseModal(false);
 
+  // <SeedPhraseVideo onClose={skip} />
```
Comment (cortisiko) [inline app/components/Views/AccountBackupStep1/index.js:218]: relating to the above

File: index.js [hunk 3]
```diff
@@ -61,6 +61,12 @@ LogBox.ignoreLogs([
   'PushNotificationIOS has been extracted', // RNC PushNotification iOS issue - https://github.com/react-native-push-notification/ios/issues/43
 ]);
 
+const IGNORE_LOGS_DEVELOPMENT = process.env.IGNORE_LOGS_DEVELOPMENT;
```
Comment (cortisiko) [inline index.js:64]: this is specifically for QA builds or regular builds? I see a benefit in having the logs during testing. Helps identify potential errors that can pollute sentry.

File: wdio/features/screen-objects/Onboarding/ImportFromSeedScreen.js [hunk 4]
```diff
@@ -67,8 +67,11 @@ class ImportFromSeed {
   }
 
   async isAlertTextVisible(text){
-    const msg = await driver.getAlertText();
-    expect(msg.includes(text)).toBe(true);
+    var msg = await driver.getAlertText();
```
Comment (cortisiko) [inline wdio/features/screen-objects/Onboarding/ImportFromSeedScreen.js:70]: Following ES6 variable declaration: the variable should be either a `const` or `let`.

File: wdio/features/screen-objects/Onboarding/OnboardingCarousel.js [hunk 5]
```diff
@@ -23,10 +24,15 @@ class WelcomeScreen {
...
 
+  get title () {
+    return Selectors.getElementByPlatform(WALLET_SETUP_SCREEN_TITLE_ID);
+  }
+
   async isScreenTitleVisible() {
     const elem = await this.splashScreenMetamaskAnimationId;
     await expect(elem).toBeDisplayed();
-    await elem.waitForDisplayed({ reverse: true });
+    // await elem.waitForDisplayed({ reverse: true });
```
Comment (cortisiko) [inline wdio/features/screen-objects/Onboarding/OnboardingCarousel.js:34]: any need for this?

**Review by cortisiko — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (Fatxx) [reply index.js:64]: That's a variable that you can set in the .js.env to hide the boxes, it's disabled by default

File: app/components/Views/AccountBackupStep1/index.js (hunk 1 above)
Comment (Fatxx) [reply app/components/Views/AccountBackupStep1/index.js:27]: This was for testing, I will put it back

File: index.js (hunk 3 above)
Comment (cortisiko) [reply index.js:64]: okay got it. Left another comment down below!

File: index.js [hunk 6]
```diff
@@ -61,6 +61,12 @@ LogBox.ignoreLogs([
...
+// Ignore box logs, usefull for QA testing in development builds
+if (IGNORE_LOGS_DEVELOPMENT === 'true') {
```
Comment (cortisiko) [inline index.js:66]: okay so I see code is added here to ignore logs. Was this intentional?

Comment (Fatxx) [reply index.js:66]: Yes, it's a condition that checks if that variable is set it hides the log boxes.

File: wdio/features/screen-objects/Onboarding/ImportFromSeedScreen.js (hunk 4 above)
Comment (Fatxx) [reply wdio/features/screen-objects/Onboarding/ImportFromSeedScreen.js:70]: This is from the @chrisleewilcox PR. I merged his PR in to this one.

Comment (chrisleewilcox) [reply wdio/features/screen-objects/Onboarding/ImportFromSeedScreen.js:70]: My bad.  Been awhile.  Changed to `let`.

**Comment by Fatxx** [comment]
Closing in favor of #5433

**Pull request closed without merging**